src/legacy/core_plugins/data/public/filter/filter_manager/filter_state_manager.ts

```diff
@@ -76,8 +74,12 @@ export class FilterStateManager {
       FilterManager.setFiltersStore(newAppFilters, FilterStateStore.APP_STATE);
       FilterManager.setFiltersStore(newGlobalFilters, FilterStateStore.GLOBAL_STATE);
 
-      this.filterManager.setFilters(newGlobalFilters.concat(newAppFilters));
-    }, 10);
+      await this.filterManager.setFilters(newGlobalFilters.concat(newAppFilters));
```

Comment:
@ppisljar I understand the shutdown, but why did you have to await on the `setFilters`.
Curious :)

Reply:
events were emitted multiple times, i don't think there was any actual issue with it just made debugging some other issues with this PR much harder.